internal/proxy/publish_handler_test.go

```diff
@@ -1 +1,206 @@
 package proxy
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"log"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/centrifugal/centrifugo/v3/internal/rule"
+
+	"github.com/centrifugal/centrifuge"
+	"github.com/centrifugal/centrifugo/v3/internal/proxyproto"
+	"github.com/centrifugal/centrifugo/v3/internal/tools"
+)
+
+type publishHandlerTestDepsConfig struct {
+	publishProxyHandler *PublishHandler
+	transport           *tools.TestTransport
+}
+
+func newPublishHandlerTestDepsConfig(proxyEndpoint string) publishHandlerTestDepsConfig {
+	proxyCfg := Config{
+		HTTPConfig: HTTPConfig{
+			Encoder: &proxyproto.JSONEncoder{},
+			Decoder: &proxyproto.JSONDecoder{},
+		},
+		PublishEndpoint: proxyEndpoint,
+	}
+
+	connectProxy, err := NewHTTPPublishProxy(
+		proxyEndpoint,
+		proxyCfg,
+	)
+	if err != nil {
+		log.Fatalln("could not create http publish proxy: ", err)
+	}
+
+	publishProxyHandler := NewPublishHandler(PublishHandlerConfig{
+		Proxy: connectProxy,
+	})
+
+	return publishHandlerTestDepsConfig{
+		publishProxyHandler: publishProxyHandler,
+		transport:           tools.NewTestTransport(),
+	}
+}
+
+func TestHandlePublishWithResult(t *testing.T) {
+	node := tools.NodeWithMemoryEngineNoHandlers()
+	defer func() { _ = node.Shutdown(context.Background()) }()
+
+	custData := "test"
+	custDataB64 := base64.StdEncoding.EncodeToString([]byte(custData))
+
+	mux := http.NewServeMux()
+	mux.HandleFunc("/publish", func(w http.ResponseWriter, req *http.Request) {
+		_, _ = w.Write([]byte(fmt.Sprintf(`{"result": {"user": "56", "expire_at": 1565436268, "b64data": "%s"}}`, custDataB64)))
```

Comment:
Looks like the data is for connect proxy ^^

Reply:
Sorry, I didn't notice)

Reply:
fixed